src/test/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorSupplierDiversityServiceImplTest.java

```diff
@@ -0,0 +1,159 @@
+        assertEquals(0, actualDiversities.size());
+    }
+    
+    @Test
+    public void testBuildSuppplierDivsersityListFromPaymentWorksVendorWithEmptyStrings() {
+        pmwVendor.setFederalDivsersityClassifications(StringUtils.EMPTY);
+        pmwVendor.setStateDivsersityClassifications(StringUtils.EMPTY);
+        
+        List<VendorSupplierDiversity> actualDiversities = supplierSerivce.buildSuppplierDivsersityListFromPaymentWorksVendor(pmwVendor);
+        assertEquals(0, actualDiversities.size());
+    }
+    
+    @Test
+    public void testBuildSuppplierDivsersityListFromPaymentWorksVendorWithNoCommas() {
+        pmwVendor.setFederalDivsersityClassifications(EIGHTA_DESCRIPTION_AFRICAN_AMERICAN);
+        pmwVendor.setStateDivsersityClassifications(WBE_DESCRIPTION);
+        
+        List<VendorSupplierDiversity> actualDiversities = supplierSerivce.buildSuppplierDivsersityListFromPaymentWorksVendor(pmwVendor);
+        assertEquals(2, actualDiversities.size());
+    }
+    
+    @Test
+    public void testBuildSuppplierDivsersityListFromPaymentWorksVendorWithInvalidOptions() {
+        pmwVendor.setFederalDivsersityClassifications("foo");
+        pmwVendor.setStateDivsersityClassifications("bar");
+        
+        List<VendorSupplierDiversity> actualDiversities = supplierSerivce.buildSuppplierDivsersityListFromPaymentWorksVendor(pmwVendor);
+        assertEquals(0, actualDiversities.size());
+    }
+    
+    @Test
+    public void testBuildSuppplierDivsersityListFromPaymentWorksVendorWithOneFederalClass() {
+        pmwVendor.setFederalDivsersityClassifications(EIGHTA_DESCRIPTION_AFRICAN_AMERICAN);
+        pmwVendor.setStateDivsersityClassifications(StringUtils.EMPTY);
+        
+        List<VendorSupplierDiversity> actualDiversities = supplierSerivce.buildSuppplierDivsersityListFromPaymentWorksVendor(pmwVendor);
+        assertEquals(1, actualDiversities.size());
+        VendorSupplierDiversity diversity = actualDiversities.get(0);
+        assertEquals(EIGHTA_CODE, diversity.getVendorSupplierDiversityCode());
+        CuVendorSupplierDiversityExtension diversityExtension = (CuVendorSupplierDiversityExtension) diversity.getExtension();
+        assertEquals(EIGHTA_CODE, diversityExtension.getVendorSupplierDiversityCode());
+    }
+    
+    @Test
+    public void testBuildSuppplierDivsersityListFromPaymentWorksVendorWithCaseDifference() {
+        pmwVendor.setFederalDivsersityClassifications("8(A) AFRICAN AMERICAN");
+        pmwVendor.setStateDivsersityClassifications(StringUtils.EMPTY);
+        
+        List<VendorSupplierDiversity> actualDiversities = supplierSerivce.buildSuppplierDivsersityListFromPaymentWorksVendor(pmwVendor);
+        assertEquals(1, actualDiversities.size());
+    }
+    
+    @Test
+    public void testBuildDateOneYearFromToday() {
+        Date actualDate = supplierSerivce.buildDateOneYearFromToday();
+        
+        Date expectedDate = new Date(Calendar.getInstance().getTimeInMillis());
```

Comment:
is this an easier way to build one year from today?

Reply:
I am not sure what you're asking here.  I am sure there are plenty of ways to build a date on year from current date.  I wanted to use two different ways in PaymentWorksVendorSupplierDiversityServiceImpl and PaymentWorksVendorSupplierDiversityServiceImplTest

Reply:
This way of building the date seemed more simple so I thought may be preferable. I wonder if it would be better if the methods would build one year from a given input date, then it might be easier to create a test that expects a specific date as output by giving a specific date as input?

Reply:
gotcha, I can do that